### PR TITLE
display tooltip for nad (line and twt)

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx [hunk 1]
```diff
@@ -159,15 +188,35 @@ function NetworkAreaDiagramContent(props) {
...
+        props.diagramId,
+        props.svgType,
+        props.svg,
+        currentNode,
+        props.loadingState,
+        diagramSizeSetter,
+        handleTogglePopover,
+    ]);
 
+    const displayTooltip = () => {
```
Comment (jamal-khey) [inline src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx:201]: i think it better for performance to not use a function to render this component

File: src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx [hunk 2]
```diff
@@ -159,15 +188,35 @@ function NetworkAreaDiagramContent(props) {
...
+        );
+    };
     /**
      * RENDER
      */
 
     return (
         <>
             <Box height={2}>{props.loadingState && <LinearProgress />}</Box>
+            {shouldDisplayTooltip && displayTooltip()}
```
Comment (jamal-khey) [inline src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx:219]:
```suggestion
            {shouldDisplayTooltip && 
            <EquipmentPopover
                studyUuid={studyUuid}
                anchorPosition={anchorPosition}
                equipmentType={hoveredEquipmentType}
                equipmentId={hoveredEquipmentId}
                loadFlowStatus={loadFlowStatus}
            />
            }
```

File: src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx [hunk 3]
```diff
@@ -122,7 +123,34 @@ function NetworkAreaDiagramContent(props) {
...
+            if (shouldDisplay) {
+                setAnchorPosition({
+                    top: mousePosition.y + 10, // Adjust for better positioning
+                    left: mousePosition.x + 10,
+                });
+                setHoveredEquipmentId(equipmentId);
+                setAnchorPosition({
+                    top: mousePosition.y + 10, // Adjust for better positioning
+                    left: mousePosition.x + 10,
+                });
```
Comment (jamal-khey) [inline src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx:145]: you call `setAnchorPosition` twice

**Review by jamal-khey — APPROVED** [review]
Tests and Code OK
- exeption on HVDC lines => fixed

**Review by YenguiSeddik — APPROVED** [review]
tests & code: ok

**Review by jamal-khey — APPROVED** [review]
Test Ok with powsyble-network-viewer 1.1.0

File: src/components/diagrams/networkAreaDiagram/network-area-diagram-content.tsx [hunk 4]
```diff
@@ -119,6 +128,7 @@ const dynamicCssRules: CSS_RULE[] = [
         thresholdStatus: THRESHOLD_STATUS.ABOVE,
     },
 ];
+const equipmentsWithPopover = ['LINE', 'TWO_WINDINGS_TRANSFORMER', 'PHASE_SHIFT_TRANSFORMER'];
```
Comment (jamal-khey) [inline src/components/diagrams/networkAreaDiagram/network-area-diagram-content.tsx:131]: use enum value is probably better
```suggestion
const equipmentsWithPopover = [
    EQUIPMENT_TYPES.LINE,
    EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER,
    FEEDER_TYPES.PHASE_SHIFT_TRANSFORMER,
];
```

[the remark this answers is not in the document]
Comment (souissimai) [reply src/components/diagrams/networkAreaDiagram/network-area-diagram-content.tsx:131]: no, not needed

**Pull request merged**